src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2221,7 +2221,10 @@ public ISession GetChildSession()
 
 				CheckAndUpdateSessionStatus();
 
-				if (_childSession == null)
+				var childImpl = _childSession as SessionImpl;
+
+				// if child session never existed or has already been disposed, create new
+				if (childImpl == null || childImpl.IsAlreadyDisposed)
```

Comment:
That's what I'm doing in the implementation of NH-3986 with a collection of child sessions, but that will come in as a separate pull request. If both end up included in the same release, this becomes moot point.